src/StreamingDataFrames/streamingdataframes/state/rocksdb/options.py

```diff
@@ -0,0 +1,66 @@
+import dataclasses
+from typing import Optional, Mapping, Literal
+
+import rocksdict
+from rocksdict import DBCompressionType
+
+from .types import RocksDBOptionsProto
+
+__all__ = ("RocksDBOptions",)
+
+CompressionType = Literal["none", "snappy", "zlib", "bz2", "lz4", "lz4hc", "zstd"]
+
+COMPRESSION_TYPES: Mapping[CompressionType, DBCompressionType] = {
+    "none": DBCompressionType.none(),
+    "snappy": DBCompressionType.snappy(),
+    "zlib": DBCompressionType.zlib(),
+    "bz2": DBCompressionType.bz2(),
+    "lz4": DBCompressionType.lz4(),
+    "lz4hc": DBCompressionType.lz4hc(),
+    "zstd": DBCompressionType.zstd(),
+}
+
+
+@dataclasses.dataclass(frozen=True)
+class RocksDBOptions(RocksDBOptionsProto):
+    """
+    Common RocksDB database options.
+
+    Please see `rocksdict.Options` for a complete description of each option.
+
+    To provide extra options that are not presented in this class, feel free
+    to override it and specify the additional values.
+    """
+
+    write_buffer_size: int = 64 * 1024 * 1024  # 64MB
```

Comment:
We use bytes here because the underlying lib also uses bytes. 
The intention was to keep the names the same with `rocksdict` to avoid confusion. 
We don't provide a full description of the params here and refer to `rocksdict` instead

Reply:
If you plan on largely being a pass-through to rocksdict, it makes sense. 

You probably saw in fluvii that some configs we tried to make a little easier to manage, because some would randomly be in kb, some in bytes, etc...so we standardized everything to MB for the user, including in the name of the configs, to make it clear what units things were in.

As such, I personally err on the side of making the names more user friendly/clear even if that means not having it be perfectly equivalent to what RocksDict uses - reason being I don't think the average user cares about what RocksDB library we're using, so ultimately we dictate what configuration looks like. For the power users who really need to get into the nitty-gritty,  you could additionally enable a 1:1 config dict passthrough (same idea for Kafka level configs, too), where they'll know we use rocksdict and thus what configs are allowed.

In the end, I don't feel strongly enough about it to push back that much on it as it's largely subjective, but I thought I'd give a little more context to my reasoning.

Reply:
In the case of these configs, they're supposed to be changed primarily by those who know how to tune RocksDB (which requires some knowledge). 
I'd rather keep the parameter names transparent to avoid ambiguity.